docs/modules/ROOT/pages/reference/extensions/google-storage.adoc

```diff
@@ -0,0 +1,60 @@
+// Do not edit directly!
+// This file was generated by camel-quarkus-maven-plugin:update-extension-doc-page
+= Google Storage
+:linkattrs:
+:cq-artifact-id: camel-quarkus-google-storage
+:cq-native-supported: true
+:cq-status: Stable
+:cq-status-deprecation: Stable
+:cq-description: Store and retrieve objects from Google Cloud Storage Service using the google-cloud-storage library.
+:cq-deprecated: false
+:cq-jvm-since: 1.9.0
+:cq-native-since: 1.9.0
+
+[.badges]
+[.badge-key]##JVM since##[.badge-supported]##1.9.0## [.badge-key]##Native since##[.badge-supported]##1.9.0##
+
+Store and retrieve objects from Google Cloud Storage Service using the google-cloud-storage library.
+
+== What's inside
+
+* xref:{cq-camel-components}::google-storage-component.adoc[Google Storage component], URI syntax: `google-storage:bucketName`
+
+Please refer to the above link for usage and configuration details.
+
+== Maven coordinates
+
+https://code.quarkus.io/?extension-search=camel-quarkus-google-storage[Create a new project with this extension on code.quarkus.io, window="_blank"]
+
+Or add the coordinates to your existing project:
+
+[source,xml]
+----
+<dependency>
+    <groupId>org.apache.camel.quarkus</groupId>
+    <artifactId>camel-quarkus-google-storage</artifactId>
+</dependency>
+----
+
+Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
+
+== SSL in native mode
+
+This extension auto-enables SSL support in native mode. Hence you do not need to add
+`quarkus.ssl.native=true` to your `application.properties` yourself. See also
+https://quarkus.io/guides/native-and-ssl[Quarkus SSL guide].
+
+== Additional Camel Quarkus configuration
+
+There are two different configuration approaches:
+
+* Google storage client can be defined via quarkus properties leveraging the Quarkiverse - Google Cloud Services - Storage (see https://github.com/quarkiverse/quarkus-google-cloud-services/blob/main/storage[extension git page]).
+Camel will autowire client into the Google-storage component.
+This configuration allows definition of only one storage client, therefore it isn't possible to define several different endpoints, which run together.
+
+* Provide client/clients for camel registry (e.g. CDI producer/bean) and reference them from endpoint.
```

Comment:
I would stress the point here like (if I understand correctly):
When multiple storage clients are needed, quarkus properties should not be used. Instead, one or more storage client bean need to be registered in the registry (e.g. via a CDI bean producer). Each storage client bean could then be referenced from distinct endpoints.

Reply:
can we open an issue on the extension to check if configuring multiple client would be an option in the future ? 

quarkus has such option for jdbc, maybe it could be possible to have the same logic here too (and probably for other extensions)

Reply:
@aldettinger Problem is unfortunately a little bit bigger here. Application properties has to be used always. Even if you define you own multiple clients.  Reason is that quarkiverse extension always creates client and if there are no properties, creation of client fails with 

>java.io.IOException: The Application Default Credentials are not available

It is unfortunate to require app properties to be filled even if it is not required, but I haven't found a way of making it worl (and to keep autowiring from camel component working) I plan to raise a ticket to quarkiverse extension asking  for optional configuration of client.

Reply:
@lburgazzoli I plan to open an issue to make configuration of client optional (see previous comment), but registration of multiple clients as you suggested makes even better sense. I'll create it in that way.

Reply:
Thanks for sharing. No big deal for the properties always needed as we'll have a separate issue.
However, I would stress that when reading the doc first, I did not understood why/when "Provide client/clients for camel registry (e.g. CDI producer/bean) and reference them from endpoint" is needed. Would you think changing the doc as below has value?
When multiple storage clients are needed, one or more storage client bean need to be registered in the registry (e.g. via a CDI bean producer). Each storage client bean could then be referenced from distinct endpoints.

Reply:
I'll rephrase the doc to provide a better explanation

Reply:
Doc is changed as requested.

Issue about multiple clients registration is created - https://github.com/quarkiverse/quarkus-google-cloud-services/issues/114
Follow-up isuue in camel-quarkus - https://github.com/apache/camel-quarkus/issues/2534